generator/src/main/java/com/scottlogic/deg/generator/violations/ViolationGenerationEngine.java

```diff
@@ -1,128 +1,61 @@
-import com.scottlogic.deg.generator.constraints.grammatical.ViolateConstraint;
 import com.scottlogic.deg.generator.generation.GenerationConfig;
-import com.scottlogic.deg.generator.outputs.manifest.ManifestWriter;
+import com.scottlogic.deg.generator.inputs.ProfileViolator;
 import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
 import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
-import com.scottlogic.deg.generator.violations.filters.ViolationFilter;
+import com.scottlogic.deg.generator.utils.FileUtils;
 
 import java.io.IOException;
-import java.nio.file.Path;
 import java.text.DecimalFormat;
 import java.util.*;
-import java.util.stream.Collectors;
 
 public class ViolationGenerationEngine implements GenerationEngine {
     private final GenerationEngine generationEngine;
-    private final Path outputPath;
-    private final ManifestWriter manifestWriter;
-    private final List<ViolationFilter> constraintsToNotViolate;
+    private final ProfileViolator profileViolator;
 
     @Inject
-    public ViolationGenerationEngine(@Named("outputPath") Path outputPath, StandardGenerationEngine generationEngine, ManifestWriter manifestWriter, List<ViolationFilter> constraintsToNotViolate){
-        this.outputPath = outputPath;
+    public ViolationGenerationEngine(ProfileViolator profileViolator,
+                                     StandardGenerationEngine generationEngine){
+        this.profileViolator = profileViolator;
         this.generationEngine = generationEngine;
-        this.manifestWriter = manifestWriter;
-        this.constraintsToNotViolate = constraintsToNotViolate;
     }
 
     @Override
     public void generateDataSet(Profile profile, GenerationConfig config, OutputTarget outputTarget) throws IOException {
-        final List<ViolatedProfile> violatedProfiles = profile.rules
-            .stream()
-            .map(rule -> getViolationForRuleTestCaseDataSet(profile, config, rule))
-            .collect(Collectors.toList());
+        List<Profile> violatedProfiles = this.profileViolator.violate(profile);
 
-        if (violatedProfiles.isEmpty()) { return; }
-
-        int filename = 1;
-        DecimalFormat intFormatter = getDecimalFormat(violatedProfiles.size());
-        try {
-            manifestWriter.writeManifest(violatedProfiles, outputPath, intFormatter, filename);
+        if (violatedProfiles.isEmpty()) {
+            return;
         }
-        catch (Exception e){}
 
-        for (ViolatedProfile violated: violatedProfiles) {
-            generationEngine.generateDataSet(violated, config,
+        DecimalFormat intFormatter = FileUtils.getDecimalFormat(violatedProfiles.size());
+
+        int filename = 1;
+        for (Profile violatedProfile : violatedProfiles) {
+            generationEngine.generateDataSet(violatedProfile, config,
                 getOutputTargetWithFilename(outputTarget, intFormatter.format(filename)));
```

Comment:
minor, i think this would read easier if there was an
    outputPath = getOutputTargetWithFilename(outputTarget, intFormatter.format(filename))
field declaration on its own line.

Reply:
I'm not convinced, I've separated out each parameter to its own line however